docs/reference/targets.md

```diff
@@ -88,6 +88,29 @@ if (this.hasResultsTarget) {
 }
 ```
 
+## Addition and Removal Callbacks
+
+Target _element callbacks_ let you respond whenever a target element is added or
+removed within the controller's element.
+
+Define a method `[name]TargetConnected` or `[name]TargetDisconnected` in the controller, where `[name]` is the name of the target you want to observe for additions or removals. The method receives the element as the first argument.
+
+Stimulus invokes each element callback any time its target elements are added or removed after `connect()` and before `disconnect()` lifecycle hooks.
+
+```js
+export default class extends Controller {
+  static targets = [ "input" ]
+
+  inputTargetConnected(element) {
+    element.classList.add("added-animation")
+  }
+
+  inputTargetDisconnected(element) {
+    element.classList.add("removed-animation")
```

Comment:
@sstephenson I now realize that the element will already be out of the DOM, so this contrived example doesn't really illustrate the point.

One use case I had in mind was sorting, would that be a better example? Can you think of a use case where having access to the `Element` instance that was removed would be helpful?